src/proof/verify.rs

```diff
@@ -107,6 +96,31 @@ impl Deref for NodeDecodingResult {
     }
 }
 
+#[inline]
+fn process_trie_node(
+    node: TrieNode,
+    walked_path: &mut Nibbles,
+    key: &Nibbles,
+) -> Result<Option<NodeDecodingResult>, ProofVerificationError> {
+    let node = match node {
+        TrieNode::Branch(branch) => process_branch(branch, walked_path, key)?,
+        TrieNode::Extension(extension) => {
+            walked_path.extend_from_slice(&extension.key);
+            if extension.child.as_hash().is_some() {
```

Comment:
@rkrasiuk this will convert the `RlpNode` into `B256` and copy it instead of just checking the length, can we introduce `is_hash`?

Reply:
yeah, that's why I suggested a is_hash fn